lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -22,54 +24,70 @@ public static function get($id) {
 
     if ($this->node && $this->node->type === 'campaign') {
       $this->variables = dosomething_helpers_get_variables('node', $this->id);
       $this->title = $this->node->title;
-      $this->tagline = $this->getTagline();
-      $this->created_at = $this->node->created;
-      $this->updated_at = $this->node->changed;
-      $this->status = $this->getStatus();
-      $this->type = $this->getType();
-      $this->time_commitment = $this->getTimeCommitment();
-
-      $this->cover_image = [
-        'default' => $this->getCoverImage(),
-        'alternate' => $this->getCoverImageAlt(),
-      ];
 
-      $this->scholarship = $this->getScholarship();
-      $this->staff_pick = $this->getStaffPickStatus();
-
-      $fact_data = $this->getFactData();
-      $this->facts = [
-        'problem' => $fact_data['fact_problem'],
-        'solution' => $fact_data['fact_solution'],
-        'sources' => $fact_data['sources'],
-      ];
-
-      $solution_data = $this->getSolutionData();
-      $this->solutions = $solution_data;
-
-      $cause_data = $this->getCauses();
-      $this->causes = [
-        'primary' => $cause_data['primary'],
-        'secondary' => $cause_data['secondary'],
-      ];
+      if ($display === 'full') {
+        $this->tagline = $this->getTagline();
+        $this->created_at = $this->node->created;
+        $this->updated_at = $this->node->changed;
+        $this->status = $this->getStatus();
+        $this->type = $this->getType();
+        $this->time_commitment = $this->getTimeCommitment();
+
+        $this->cover_image = [
+          'default' => $this->getCoverImage(),
+          'alternate' => $this->getCoverImageAlt(),
+        ];
+
+        $this->scholarship = $this->getScholarship();
+        $this->staff_pick = $this->getStaffPickStatus();
+
+        $fact_data = $this->getFactData();
+        $this->facts = [
+          'problem' => $fact_data['fact_problem'],
+          'solution' => $fact_data['fact_solution'],
+          'sources' => $fact_data['sources'],
+        ];
+
+        $solution_data = $this->getSolutionData();
+        $this->solutions = $solution_data;
+
+        $cause_data = $this->getCauses();
+        $this->causes = [
+          'primary' => $cause_data['primary'],
+          'secondary' => $cause_data['secondary'],
+        ];
+
+        $action_types_data = $this->getActionTypes();
+        $this->action_types = [
+          'primary' => $action_types_data['primary'],
+          'secondary' => $action_types_data['secondary'],
+        ];
+
+        $this->issue = $this->getIssue();
+        $this->tags = $this->getTags();
+
+        $timing = $this->getTiming();
+        $this->timing = $timing;
+      }
 
-      $action_types_data = $this->getActionTypes();
-      $this->action_types = [
-        'primary' => $action_types_data['primary'],
-        'secondary' => $action_types_data['secondary'],
+      $reportback_info = $this->getReportbackInfo();
+      $this->reportback_info = [
```

Comment:
The difference that I think this does make is that if `copy` or `noun` or whatever is actually not set in the return from `getReportbackInfo()`, then `$this->reportback_info` has that property set to... NULL?